core-plugins/src/test/java/co/cask/hydrator/plugin/batch/aggregator/function/CollectSetTest.java

```diff
@@ -0,0 +1,74 @@
+package co.cask.hydrator.plugin.batch.aggregator.function;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Set;
+
+/**
+ * @author Harsh Takkar
+ */
+public class CollectSetTest {
+
+  @Test
+  public void testIntCollectSet() {
+    Schema schema = Schema.recordOf("test", Schema.Field.of("x", Schema.of(Schema.Type.INT)));
+    CollectSet collectSet = new CollectSet("x", Schema.of(Schema.Type.INT));
+    collectSet.beginFunction();
+    collectSet.operateOn(StructuredRecord.builder(schema).set("x", 1).build());
+    collectSet.operateOn(StructuredRecord.builder(schema).set("x", 2).build());
+    collectSet.operateOn(StructuredRecord.builder(schema).set("x", 1).build());
+    Set set = collectSet.getAggregate();
+    Assert.assertArrayEquals(new  Integer[] {1, 2}, set.toArray());
```

Comment:
Should be comparing Sets instead of array comparison, as the order is not guaranteed.

Reply:
Corrected